Yosemite/Yosemite/PointOfSale/POSItemProvider.swift

```diff
@@ -0,0 +1,14 @@
+public protocol POSItem {
+    var itemID: UUID { get }
+    var productID: Int64 { get }
+    var name: String { get }
+    var price: String { get }
+    var imageURL: URL { get }
+    var details: [String] { get }
+
+    func makeCartItem() -> CartItem
```

Comment:
❓ this doesn't seem used in the PR, is this necessary? It looks like the `CartItem` is created in `PointOfSaleDashboardViewModel.addItemToCart`.

If this isn't needed now, maybe we can move `CartItem` back to the app layer, because the POSItemProvider doesn't need to know about the cart. If we need the cart item in the data layer (Yosemite for now) (possibly when syncing order), we can create another version in Yosemite for what's needed.

Reply:
These are not yet necessary, and doesn't seem that provider needs to know about them indeed. I've removed the additional properties and moved it back to app layer on https://github.com/woocommerce/woocommerce-ios/pull/12882/commits/e8d12cd335c3a3836f271d1e2731aa44bf6c2db1 and https://github.com/woocommerce/woocommerce-ios/pull/12882/commits/4754c81d66ce3627543aae8f8e74d5d53e96b768